src/crons/status.checker/status.checker.service.ts

```diff
@@ -0,0 +1,444 @@
+          this.apiStatusMetricsService.setTotalShardRounds(shardId, round);
+          this.apiStatusMetricsService.setTotalShardNonces(shardId, nonce);
+        }
+      });
+    }, true);
+  }
+
+  @Cron('*/6 * * * * *')
+  async handleMexEconomicValues() {
+    await Locker.lock('MexEconomics values', async () => {
+      await this.lock.acquire('MexEconomics values', async () => {
+        const economics = await this.mexEconomicService.getMexEconomics();
+        for (const [key, value] of Object.entries(economics)) {
+          this.apiStatusMetricsService.setMexEconomicsValue(key, value);
+        }
+      });
+    }, true);
+  }
+
+  @Cron('*/6 * * * * *')
+  async handleEconomicValues() {
+    await Locker.lock('Economics values', async () => {
+      await this.lock.acquire('economics values', async () => {
+        const economics = await this.economicService.getEconomics();
+        for (const [key, value] of Object.entries(economics)) {
+          this.apiStatusMetricsService.setEconomicsValue(key, value);
+        }
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  async handleTokensCounterValues() {
+    await Locker.lock('Tokens Counter Metrics', async () => {
+      await this.lock.acquire('tokens counter', async () => {
+        const tokens = await this.tokenService.getTokenCount({});
+        const performanceProfiler = new PerformanceProfiler();
+
+        performanceProfiler.stop();
+
+        if (tokens < 1619) {
+          this.apiStatusMetricsService.checkTokensCountValue('success', performanceProfiler.duration);
+        } else {
+          this.apiStatusMetricsService.checkTokensCountValue('error', performanceProfiler.duration);
+        }
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  async handleProvidersValues() {
+    await Locker.lock('Providers Counter Metrics', async () => {
+      await this.lock.acquire('providers counter', async () => {
+        const providers = await this.providerService.getAllProviders();
+        const performanceProfiler = new PerformanceProfiler();
+
+        performanceProfiler.stop();
+
+        if (providers.length < 133) {
+          this.apiStatusMetricsService.setProvidersCountValue('success', performanceProfiler.duration);
+        } else {
+          this.apiStatusMetricsService.setProvidersCountValue('error', performanceProfiler.duration);
+        }
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_SECONDS)
+  async handleTokensSupplyInvalidations() {
+    await Locker.lock('Tokens supply invalidations', async () => {
+      await this.lock.acquire('tokens supply', async () => {
+        const performanceProfiler = new PerformanceProfiler();
+        const tokens = await this.tokenService.getTokens(new QueryPagination({ size: 1000 }), new TokenFilter());
```

Comment:
why are we only checking the first 1000 tokens?

Reply:
Already discussed